docs/docusaurus/docs/conceptual_guides/metricproviders.md

```diff
@@ -0,0 +1,105 @@
+---
+sidebar_label: 'MetricProviders'
+title: 'MetricProviders'
+id: metricproviders
+description: Learn how MetricProviders are an integral component of the Expectation software development kit (SDK).
+---
+
+MetricProviders generate and register Metrics to support Expectations, and they are an important part of the Expectation software development kit (SDK). Typically, MetricProviders are used to resolve complex requirements and are not implemented widely. If you're considering using MetricProviders, you're encouraged to join the [Great Expectations Slack community](https://greatexpectations.io/slack) and discuss your requirements in the [#gx-community-support](https://greatexpectationstalk.slack.com/archives/CUTCNHN82) channel.
+
+You'll encounter references to MetricProviders in many of the topics in the Great Expectations (GX) documentation. An in-depth knowledge of MetricProvider functionality is helpful, but not required to successfully complete the tasks documented in these topics. However, some knowledge of MetricProvider functionality is useful if you're developing Custom Expectations, or you're maintaining or extending MetricProvider classes.
+
+MetricProviders let you declare all the Metrics that are needed to support an Expectation in concise, Don't Repeat Yourself (DRY) syntax. MetricProviders support the following use cases:
+
+- Conceptually grouping code for Metrics calculation across multiple backends.
+- Incremental development. You can develop Metrics for each Execution Engine, one at a time.
+- Sharing Metrics across Expectations.
+- Generating Metrics that can be inferred from specific types of Expectation automatically.
+
+## Assumed knowledge
+
+To get the most out of the information provided here, you should have an understanding of:
+
+- Expectations
+- Metrics
+- The Metric registry
+- ExecutionEngines
+
+## When MetricProviders are required
+
+A minimum of one supporting Metric is required by every Expectation. For example, `expect_column_mean_to_be_between` relies on a Metric that calculates the mean of a column. Often, an Expectation requires multiple Metrics. For example, the following Metrics are required in the `expect_column_values_to_be_in_set` Expectation:
+
+- `column_values.in_set.unexpected_count`
+- `column_values.in_set.unexpected_rows`
+- `column_values.in_set.unexpected_values`
+- `column_values.in_set.unexpected_value_counts`
+
+To allow Expectations to work with multiple backends, methods for calculating Metrics need to be implemented for each ExecutionEngine. For example the `aggregate_fn` utility that is used with SQL and Spark requires the `column_values.in_set.unexpected_count.aggregate_fn` Metric. 
```

Comment:
I think this would be a stronger example if it were explaining the different methods that are used to calculate the same metric in different ExecutionEngines.  Something along the lines of: "For example, the `some_metric` in `ExpectSomethingToBeTrue` is implemented with the `some_spark_method` utility for the Spark ExecutionEngines, but the pandas ExecutionEngine calculates the metric using `something_different`.

Reply:
You'll probably need feedback from someone with more engineering knowledge to put together that kind of an example, but I think it would be more meaningful given the topic sentence here.

Reply:
@austiezr could you provide the information necessary to update the text as Rachel has recommended, or could you provide the name of a contact who can provide the necessary information?

Reply:
This `column_mean` [Metric](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/expectations/metrics/column_aggregate_metrics/column_mean.py) is a good example -- `_pandas` is implemented by calling the Pandas-native `.mean()` method on the column, whereas `_spark` is implemented with a Spark-native mean function & `_sqlalchemy` is implemented with a SQLAlchemy generic function.